### DISCUSSION TITLE
Is it possible to install waybar-hyprland with flake?

### DISCUSSION THREAD

**Discussion by woojiq** [hyprwm/Hyprland#1729] [Q&A] [answered]
I read the **hyprland wiki** and successfully installed **hyprland** using _nix flakes_. I also want to install the latest revision of **waybar** (there is one critical bug in the last stable release 0.9.17). I saw that you mention your custom waybar (waybar-hyprland) in `flake.nix`. Is it possible to do?
Why I thought this is possible: [NixOs Flake Store](https://search.nixos.org/flakes?channel=unstable&show=waybar-hyprland&from=0&size=50&sort=relevance&type=packages&query=waybar) 
+ 
https://github.com/hyprwm/Hyprland/blob/a85a6fa6c8ff9e51ba774cec112b408bce601451/flake.nix#L85-L91
What I tried:
1. 
``` nix
hyprland.url = "github:hyprwm/hyprland#hyprland";
```
Error: `error: unexpected fragment 'hyprland' in flake reference 'github:hyprwm/hyprland#hyprland'`
2.
``` nix
hyprland.url = "github:hyprwm/hyprland";
...
wayland-hyprland.enable = true;
```
Error: `error: The option 'home-manager.users.woojiq.programs.waybar-hyprland' does not exist....`
Btw, thank you for the great window manager. It is honey for the eyes after sway :)

**UPD**: 
And just to make sure I'm right. Is this the correct way to enable `nvidiaPatches`:
``` nix
programs.hyprland = {
      enable = true;
      nvidiaPatches = true;
};
```
This compiles without error and works well. I just want to be sure)

**UPD.2.0.**: Does this mean that `waybar-hypland` overwrites `waybar`, so when I type `pkgs.waybar` it installs the patched one?

**UPD.3.0.**. There is no "bug". Owners, feel free to delete this post as it's useless :smiling_face_with_tear:

**Comment by woojiq** [comment, marked as answer]
<details><summary>Old solution with overlay:</summary>

I did it. I am new to the nix ecosystem and don't know all features. I don't know exactly why, but I needed to write overlay despite one being written in hyprland flake. I copy the overlay from this [repo](https://github.com/ocfox/nixos-config/blob/master/overlays/default.nix), it's the same as in the hyprland flake.
</details>

<details><summary>One more old solution</summary>

**New solution with existing flake overlay:**
As I mentioned above there is the waybar overlay with the patch to use wlr/workspaces. To use it and not write patch yourself you can override existing waybar package like this:
``` nix
{ hyprland, ... }:
...
let
  system = "x86_64-linux";
in
...
pkgs = import nixpkgs {
  inherit system;
  config.allowUnfree = true;
  overlays = [ (final: prev: { waybar = hyprland.packages.${system}.waybar-hyprland; }) ];
};
```
Corresponding lines in the hyprland flake: 
https://github.com/hyprwm/Hyprland/blob/71ef1bde7e45f94ceaab925ae065f32f91510890/flake.nix#L106-L110
</details>

**Nice and easy:**
```nix
programs.waybar.package = inputs.hyprland.packages.${system}.waybar-hyprland;
```
P.S. I originally asked this question as a noob, but now I realize that the answer is so easy that basically no one needs this discussion, I was just dumb and coming up with complex solutions)

**Reply by KuSi833** [reply]
Cheers for documenting this, was helpful.

**Reply by webbb82** [reply]
Gentlemen, He's done it!!!!   ```slow clap  ```slow clap ```slow clap with` headknod
